carp.deployments.core/src/commonMain/kotlin/dk/cachet/carp/deployments/application/ParticipationService.kt

```diff
@@ -42,13 +42,13 @@ interface ParticipationService : ApplicationService<ParticipationService, Partic
     suspend fun getParticipantDataList( studyDeploymentIds: Set<UUID> ): List<ParticipantData>
 
     /**
-     * Set participant [data] for the given [inputDataType] in the study deployment with [studyDeploymentId].
+     * Set participant [data] in the study deployment with [studyDeploymentId].
      *
      * @throws IllegalArgumentException when:
      *   - there is no study deployment with [studyDeploymentId]
-     *   - [inputDataType] is not configured as expected participant data in the study protocol
-     *   - [data] is invalid data for [inputDataType]
+     *   - one or more of the keys in [data] isn't configured as expected participant data in the study protocol
+     *   - one or more of the set [data] isn't valid for the corresponding input data type
      * @return All data for the specified study deployment, including the newly set data.
      */
-    suspend fun setParticipantData( studyDeploymentId: UUID, inputDataType: InputDataType, data: Data? ): ParticipantData
+    suspend fun setParticipantData( studyDeploymentId: UUID, data: Map<InputDataType, Data?> ): ParticipantData
```

Comment:
Is it a necessary constrain to only allow one of each `InputDataType` ?

Reply:
Yes, this is by design and I believe where some of the confusion came from. `inputDataType` is supposed to be unique. That's why for custom fields (which share the same _data_ type), a unique `inputDataType` is generated.

The idea here is that what fully defines the data to be requested, e.g. "biological sex", is unique. This does presume data is requested _for one specific participant_. But, we currently take shortcuts here and have not yet designed multiple participants into core, though we do know where to fit it in afterwards.

Reply:
Maybe an indicator this should be named differently? Or, documented better?

Reply:
> Maybe an indicator this should be named differently? Or, documented better?
It's more like `InputDataId` since that's really what it is.

Reply:
That would be slightly inconsistent with `typealias DataType = NamespacedId`, however. I think the point here is that it's _usage_ here is as an _id_, but it also is a `DataType`.

Reply:
I think `InputDataId` has its own problems (elsewhere in the code where this is used). Instead, I just updated the documentatoin of `InputDataType` slightly. I think any confusion here will best be addressed through documentation.